scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -59,6 +59,38 @@ def test_quat_double_to_canonical_single_cover():
     assert_array_almost_equal(r.as_quat(canonical=True), expected_quat)
 
 
+def test_quat_double_cover():
+    # See the Rotation.from_quat() docstring for scope of the quaternion
+    # double cover property.
+    # Check from_quat and as_quat(canonical=False)
+    q = np.array([0, 0, 0, -1])
+    r = Rotation.from_quat(q)
+    assert all(q == r.as_quat(canonical=False))
```

Comment:
Similar to what you did below, I suspect `assert_equal` from NumPy will give you a better error message if something goes wrong because of the stats, etc.:

```
E   AssertionError: 
E   Arrays are not equal
E   
E   Mismatched elements: 1 / 4 (25%)
E   Max absolute difference: 3.
E   Max relative difference: 0.
E    x: array([ 3,  0,  0, -1])
E    y: array([ 0.,  0.,  0., -1.])
```

Reply:
Makes sense, switched to that. These are floats but there should be no operations being performed on this data here so an exact test is appropriate